Modules/sha256module.c

```diff
@@ -681,43 +681,44 @@ static struct PyMethodDef SHA_functions[] = {
     {NULL,      NULL}            /* Sentinel */
 };
 
-
-/* Initialize this module. */
-
-static struct PyModuleDef _sha256module = {
-        PyModuleDef_HEAD_INIT,
-        "_sha256",
-        NULL,
-        -1,
-        SHA_functions,
-        NULL,
-        NULL,
-        NULL,
-        NULL
-};
-
-PyMODINIT_FUNC
-PyInit__sha256(void)
+static int sha256_exec(PyObject *module)
 {
-    PyObject *m;
-
     Py_SET_TYPE(&SHA224type, &PyType_Type);
     if (PyType_Ready(&SHA224type) < 0) {
-        return NULL;
+        return -1;
     }
     Py_SET_TYPE(&SHA256type, &PyType_Type);
     if (PyType_Ready(&SHA256type) < 0) {
-        return NULL;
+        return -1;
     }
 
-    m = PyModule_Create(&_sha256module);
-    if (m == NULL)
-        return NULL;
-
     Py_INCREF((PyObject *)&SHA224type);
-    PyModule_AddObject(m, "SHA224Type", (PyObject *)&SHA224type);
+    PyModule_AddObject(module, "SHA224Type", (PyObject *)&SHA224type);
```

Comment:
`Unlike other functions that steal references, PyModule_AddObject() only decrements the reference count of value on success.` from https://docs.python.org/3/c-api/module.html?highlight=pymodule_addobject#c.PyModule_AddObject

so you need do something like:
```
Py_INCREF((PyObject *)&SHA224type);
if (PyModule_AddObject(module, "SHA224Type", (PyObject *)&SHA224type) < 0) {
    Py_DECREF(PyObject *)&SHA224type);
    return -1;
}
```

Reply:
@shihai1991 does this mean there was a possible leak pre-existing before my changes?

Reply:
Yes, you are right. you can fix this leak in your PR ;)